lib/api/controllers/securityController.js

```diff
@@ -388,7 +390,19 @@ module.exports = function SecurityController (kuzzle) {
 
         return kuzzle.repositories.user.load(newRequestObject.data._id)
           .then(user => {
-            return kuzzle.repositories.user.persist(_.extend(user, newRequestObject.data.body), { database: { method: 'update' } });
```

Comment:
Nope, the `Repository.load` method didn't return a real `User` object: it contains data in a `_source` subfield (seems to be the raw output of elastic search) so the `_.extend` didn't work as we want here.

On my way, we need to plan a refactor on the Repositories methods to ensure the validity of theirs return values.

To reproduce the bug, just create a new user with admin profile, update it with anonymous profile. Then get it and see that the profile is switched to default.